resources/assets/js/features/bookings/components/BookingsContainer.js

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import BookingService from 'app/services/BookingService';
+import Preloader from 'app/components/Preloader';
+import {Pagination} from 'app/components/Pagination';
+import BookingItem from './BookingItem';
+import '../styles/bookings-container.scss';
+
+class BookingsContainer extends React.Component {
+    constructor() {
+        super();
+        this.state = {
+            page: 1,
+            limit: 10,
+            data: [],
+            meta: {},
+            errors: {},
+            preloader: false
+        };
+
+        this.onChangePage = this.onChangePage.bind(this);
+    }
+
+    componentWillMount() {
+        this.getData(this.props, this.state);
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (nextProps.filter !== this.props.filter) {
+            this.setState({page: 1});
+            this.getData(nextProps, Object.assign(this.state, {page: 1}));
+        } else {
+            this.getData(nextProps, this.state);
+        }
+    }
+
+    getData(props, state) {
+        const { filter } = props,
+            {page, limit} = state;
+
+        this.setState({preloader: true});
+
+        BookingService.getBookingsList(filter, page, limit)
+            .then(data => this.setState({
+                data: data.data,
```

Comment:
use global state to store data from server, you can set local state based on global one, but main data source in this case should be global